lib/collaboration.php

```diff
@@ -0,0 +1,204 @@
+<?php
+/**
+ * This is collaboration route for set and get of key value of RTC signals.
+ *
+ * @package gutenberg
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	die( 'Silence is golden.' );
+}
+
+add_action( 'rest_api_init', 'collaboration_routes' );
+
+function collaboration_routes() {
+  $current_user = wp_get_current_user();
+
+  register_rest_route( 'collaborate', '/username', array(
+    'methods'  => 'GET',
+    'callback' => 'get_username',
+    'args'     => array(
+      'username' => $current_user->user_login
+    )
+  ));
+
+  register_rest_route( 'collaborate', '/get/(?P<key>\w+.*)', array(
+    'methods'  => 'GET',
+    'callback' => 'collaboration_get',
+  ));
+
+  register_rest_route( 'collaborate', '/set', array(
+    'methods'  => 'POST',
+    'callback' => 'collaboration_set',
+  ));
+
+  register_rest_route( 'collaborate', '/remove', array(
+    'methods'  => 'POST',
+    'callback' => 'collaboration_remove',
+  ));
+}
+
+/**
+ * Returns the username who is logged in.
+ *
+ * @param object $data as request data along with args.
+ * @return bool else username if logged in.
+ */
+function get_username( $data ){
+  $username = $data->get_attributes();
+  return $username['args']['username'];
+}
+
+/**
+ * Check if string is valid base64.
+ * 
+ * @param string base64
+ * @return bool if its valid
+ */
+function base64_check( $string ) {
```

Comment:
This is a rather generic function that might need consideration as part of WP utility functions. cc @pento @mdawaffe

Reply:
Why are the extra sanity checks needed over the strict flag in `base64_decode()`?

Reply:
Updated.